interview-questions.md

```diff
@@ -8,36 +8,47 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: I think the new key that I am going to make will be in the student model because the student model goe back to the cohort and not the opposite. now regarding how to make the new key I am not sure, but my best shot would be making it by making a rout. 
 
-Researched answer:
+Researched answer: right way to fix this is to migrate a new key data for the student model. 
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: show, new, update, I belive because in these i can put the new data i want.
 
-Researched answer:
+Researched answer:  index, new, show, and edit must pass params because we can identify what is included 
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: I am very familiar with rails generate model,  where it creates new data and column names to be made and what is in each column.
 
-Researched answer:
+Researched answer: rails generate migration action_name, is what allows me to make changes to all that data.  rails generate resource, creates a model and all RESTful routes within. rails generate model,  where it creates new data and column names to be made and what is in each column.
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET" location: /students
+action: "GET" location: /students- index
 
-action: "POST" location: /students
+action: "POST" location: /students- index
 
-action: "GET" location: /students/new
+action: "GET" location: /students/new- create a new student in a form
 
-action: "GET" location: /students/2
+action: "GET" location: /students/2-show information about one student with the id (2)
 
-action: "GET" location: /students/2/edit
+action: "GET" location: /students/2/edit-edit student with id 2
 
-action: "PATCH" location: /students/2
+action: "PATCH" location: /students/2-update
 
-action: "DELETE" location: /students/2
+action: "DELETE" location: /students/2-destroy
```

Comment:
I would recommend reviewing the [RESTful routes](https://github.com/learn-academy-2022-delta/Syllabus/blob/main/rails/restful-routes-crud.md) section in the syllabus. This has a really great table that shows what each one of these does.  Also, make sure to answer both questions.